handlers/infobip/infobip.go

```diff
@@ -1,3 +1,124 @@
 package infobip
 
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+)
+
 /* no logs */
+
+func init() {
+	courier.RegisterHandler(NewHandler())
+}
+
+type handler struct {
+	handlers.BaseHandler
+}
+
+// NewHandler returns a new Infobip handler
+func NewHandler() courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("IB"), "Infobip")}
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	err := s.AddReceiveMsgRoute(h, "POST", "receive", h.ReceiveMessage)
+	return err
+}
+
+// ReceiveMessage is our HTTP handler function for incoming messages
+func (h *handler) ReceiveMessage(channel courier.Channel, w http.ResponseWriter, r *http.Request) ([]courier.Msg, error) {
+	ie := &infobipEnvelope{}
+	err := handlers.DecodeAndValidateJSON(ie, r)
+	if err != nil {
+		return nil, err
+	}
+
+	if ie.MessageCount == 0 {
+		return nil, courier.WriteIgnored(w, r, "Ignoring request, no message")
+	}
+
+	msgs := []courier.Msg{}
+	if len(ie.Results) > 0 {
+		for index, infobipMessage := range ie.Results {
+			fmt.Println(index)
+			messageID := infobipMessage.MessageID
+			text := infobipMessage.Text
+			//receiver := infobipMessage.To
+			sender := infobipMessage.From
+			dateString := infobipMessage.ReceivedAt
+
+			date := time.Now()
+			if dateString != "" {
+				date, err = time.Parse("2006-01-02T15:04:05.999999999-0700", dateString)
+				if err != nil {
+					date = time.Now()
+				}
+			}
+			date = date.UTC()
+
+			// create our URN
+			urn := courier.NewTelURNForChannel(sender, channel)
```

Comment:
Would just not declare a local variable for sender above and do this inline instead